### ISSUE TITLE
Linux AppImage won't open on Linux / Steam Deck

### ISSUE THREAD

**Issue by GreenNinja2525** [Lime3DS/Lime-3DS-Emulator#127]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

2110

### Description of Issue

No matter what I tried, the AppImage refuses to open. Version 2109 seems to open just fine. Other AppImages work perfectly normal, but this version will not open, not even through the terminal or by adding it to steam itself.

### Expected Behavior

Open AppImage and show behavior like version 2109

### Reproduction Steps

Use a Steam Deck, download the latest version (2110) and attempt to open the AppImage and watch it not open

### Log File

I can't get one to generate

### System Configuration

CPU: Zen 2
GPU/Driver: AMD APU
RAM: 16GB LPDDR5 (vram is set at 4GB)
OS: Steam Os on Steam Deck
AppImage downloaded from Github

**Comment by busywait** [comment]
Me too. Starting from a terminal prompt in Desktop mode shows missing libraries on the Steam Deck (SteamOS 3.5.17):
```
(deck@steamdeck Applications)$ ./lime-qt.AppImage 
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/AppRun.wrapped)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/AppRun.wrapped)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libsndio.so.7)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libQt6Gui.so.6)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libQt6Gui.so.6)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libQt6Network.so.6)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/checkrt/cxx/libstdc++.so.6)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libdbus-1.so.3)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libbsd.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libglib-2.0.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libxkbcommon.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libgssapi_krb5.so.2)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libicui18n.so.74)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libicui18n.so.74)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libicuuc.so.74)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libicuuc.so.74)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libpulsecommon-16.1.so)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libsystemd.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libkrb5.so.3)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libk5crypto.so.3)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libkrb5support.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libgomp.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libsndfile.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libm.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libsndfile.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libapparmor.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libcap.so.2)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libgcrypt.so.20)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libgio-2.0.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libgnutls.so.30)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libldap.so.2)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/liblber.so.2)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libmount.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libselinux.so.1)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libcrypto.so.3)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libp11-kit.so.0)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libsasl2.so.2)
/tmp/.mount_lime-qlTTWlc/AppRun.wrapped: /usr/lib/libc.so.6: version `GLIBC_2.38' not found (required by /tmp/.mount_lime-qlTTWlc/usr/bin/../lib/libblkid.so.1)
```

**Comment by busywait** [comment]
Just for fun, I put my Steam Deck in to developer mode and selected the "main" OS update channel (work in progress that might end up being part of the next OS release). That version of SteamOS does allow lime-qt.AppImage to run.

**Comment by OpenSauce04** [comment]
Although this is an issue that will solve itself with time as more people recieve glibc 2.38, at the moment it seems this version is a bit too new to be considered a minimum
I will look into reducing to minimum required glibc version for the next update. For now just use the Flatpak

**Comment by rtiangha** [comment]
I'm not sure if the AppImage container can be configured to build explicit static executables, but if it can, that might be another option; that used to be the traditional way to get around glibc incompatibilities for distribution; i.e. build the library into the binary, rather than rely on dynamic linking with a user's system library.

**Comment by OpenSauce04** [comment]
I've read online that statically linking against glibc can cause some funky issues with GPU drivers, so it may be best to avoid that

**Comment by OpenSauce04** [comment]
This has been addressed for the next update. We now build against glibc 2.36

**Comment by farvardin** [comment]
@OpenSauce04 I'm still on Linux Mint 21.3 / Ubuntu 22.04 which is still supported until 2026 and it's using libc 2.35.
I suppose it would be a sensitive choice to stay on a version below 2.35, if it's possible, as many users are still using those versions. Appimages are meant to be usable by most people. I have around 70 AppImages on my system and Lime3DS is the first one which is using so much recent libc version.

**Comment by OpenSauce04** [comment]
Long term support doesn't mean that other developers have to support it, it means that it will continue being supported by the distro developers.

Our long term plan is to stay in step with the most recent Debian release's glibc version, which as of now is 2.36.

I would recommend simply updating to Linux Mint 22, there isn't really a reason not to.

**Comment by directentis1** [comment]
@farvardin I agree, AppImage should have the widest compatibility spectrum,..  otherwise one can use Debian package or flatpak.
Ubuntu 22.04 is still very popular, and 24.04 was only recently released.

I tried to downgrade debian:12-slim (Bookworm) to debian:11-slim (Bullseye), but no luck with dependency hell... first is Qt6 (which Bullseye already has backport), then clang and CMake.

**Comment by farvardin** [comment]
@directentis1 thank you for trying... for my part I've compiled Lime3DS from source, it wasn't that difficult after all.

**Comment by rtiangha** [comment]
If you wanted to build your own version through GitHub Actions, what you could do is clone the Lime3DS repository and replace the `linux` entry in `.github/workflows/build.yml` with this:

```
  linux:
    runs-on: ubuntu-22.04
    strategy:
      matrix:
        target: ["appimage", "fresh"]
    env:
      CCACHE_DIR: ${{ github.workspace }}/.ccache
      CCACHE_COMPILERCHECK: content
      CCACHE_SLOPPINESS: time_macros
      OS: linux
      TARGET: ${{ matrix.target }}
    steps:
      - uses: actions/checkout@v4
        with:
          submodules: recursive
      - name: Set up cache
        uses: actions/cache@v4
        with:
          path: ${{ env.CCACHE_DIR }}
          key: ${{ runner.os }}-${{ matrix.target }}-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-${{ matrix.target }}-
      - name: Install Clang PPA 
        uses: myci-actions/add-deb-repo@11
        with:
          repo: deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main
          repo-name: llvm
          keys-asc: https://apt.llvm.org/llvm-snapshot.gpg.key
      - name: Get latest CMake and Ninja
        uses: lukka/get-cmake@latest
        with:
          cmakeVersion: latest
          ninjaVersion: latest 
      - name: Install New Packages
        uses: gerlero/apt-install@v1
        with:
          packages: ccache clang-18 clang-tools-18 clang-18-doc libclang-common-18-dev libclang-18-dev libclang1-18 clang-format-18 python3-clang-18 clangd-18 clang-tidy-18 clang-format-18 libasound-dev xorg-dev libx11-dev libxext-dev jackd libpipewire-0.3-dev libsndio-dev libxcb-cursor-dev libasound2-dev libpulse-dev libaudio-dev libjack-dev libsndio-dev libx11-dev libxext-dev libxrandr-dev libxcursor-dev libxfixes-dev libxi-dev libxss-dev libwayland-dev libxkbcommon-dev libdrm-dev libgbm-dev libgl1-mesa-dev libgles2-mesa-dev libegl1-mesa-dev libdbus-1-dev libibus-1.0-dev libudev-dev fcitx-libs-dev
      - name: Update System Packages
        run: sudo apt-get update && sudo apt-get -y dist-upgrade && sudo apt-get clean
      - name: Install Qt
        uses: jurplel/install-qt-action@v4
        with:
          aqtversion: '==3.1.*'
          version: '6.7.3'
          host: 'linux'
          target: 'desktop'
          arch: 'linux_gcc_64'
          modules: 'qtmultimedia qtwaylandcompositor'
          cache: 'true'
      - name: Build
        run: ./.ci/linux.sh
      - name: Pack
        run: ./.ci/pack.sh
        if: ${{ matrix.target == 'appimage' }}
      - name: Upload
        uses: actions/upload-artifact@v4
        if: ${{ matrix.target == 'appimage' }}
        with:
          name: ${{ env.OS }}-${{ env.TARGET }}
          path: artifacts/
```

It'll build an AppImage based off of Ubuntu 22.04 and it's what I use for my own custom builds to avoid using a hosted Docker container so that I can ensure I'm using up-to-date compilers and dependencies at build time.

Edit: You could also try replacing the runner with Ubuntu 20.04 although I've never tried it myself and don't know if it would work (you'll need to replace all the `jammy` entries in the Clang PPA with `focal`, though).

**Comment by directentis1** [comment]
> If you wanted to build your own version through GitHub Actions, what you could do is clone the Lime3DS repository and replace the `linux` entry in `.github/workflows/build.yml` with this:
> ....

Got it, at first I was a bit stuck with the Debian container, so I had to build it my own way, but then it crashed with Cmake and Clang when running build.yml (because Bullseye uses a too old version of Cmake, at least compared to Ubuntu 22.04)

https://github.com/directentis1/lime3ds-build/blob/master/Dockerfile

---
I have built it by your help.
https://github.com/directentis1/Lime3DS/actions/runs/11093466745

It's strange that the AppImage version here is much lighter than the one I built manually 😁

**Comment by directentis1** [comment]
> @directentis1 thank you for trying... for my part I've compiled Lime3DS from source, it wasn't that difficult after all.

That's my last resort, haha, it works but there are some problems with Vulkan and Nvidia (I'm not sure why, even though I followed the [instructions](https://github.com/Lime3DS/Lime3DS/wiki/Building-From-Source) exactly).
I built it with Ubuntu 22.04 (Google cloud shell).

![image](https://github.com/user-attachments/assets/61c1641b-bec4-4d5c-a0b1-8b9c8ef106f5)
![image](https://github.com/user-attachments/assets/b94bd34e-a9b2-4e7f-8335-1503b7c06e41)

**Comment by rtiangha** [comment]
Did that segfault happen with the version you built off of GitHub? Could you post your logs? Also, one thing you could try is lowering the version of QT; it's at 6.7.3 in the thing I posted above, but any version of the form 6.x.3 can work. I'm curious to see if the crash still happens on older QT6.

And I don't know if this will help with the Vulkan issues, but you could also try adding the Vulkan SDK to your build before compiling. Insert this snippet before Install New Packages:

```
      -name: Install Vulkan SDK
       uses: humbletim/install-vulkan-sdk@v1.1.1
       with:
         version: latest
         cache: true
```
And I don't know if Ubuntu has such a thing as a Nvidia dev package or library, but if it does, perhaps installing that too might help.

**Comment by directentis1** [comment]
> Did that segfault happen with the version you built off of GitHub? Could you post your logs? Also, one thing you could try is lowering the version of QT; it's at 6.7.3 in the thing I posted above, but any version of the form 6.x.3 can work. I'm curious to see if the crash still happens on older QT6.

I encountered both versions, the first built manually (installing each specific dependency package according to [the instructions](https://github.com/Lime3DS/Lime3DS/wiki/Building-From-Source)); the second one builds automatically with GH's action.

It seems that it only happens when using Intel based GPU, though I don't encounter it when using it on a machine with an Nvidia card.

It seems that the above segmentation error only occurs when closing an open window, otherwise when loading the game, everything still works fine, although there is a bit of bottleneck when transitioning scenes/cutscene (at least when I tried it with Pokémon Omega).

![image](https://github.com/user-attachments/assets/1e8eae3d-d201-4e9c-939f-283c29d3ec65)

I guess 'cause Vulkan doesn't work, Lime automatically fallback to OpenGL, and gets the above error.

![image](https://github.com/user-attachments/assets/9131d6ac-0852-40c9-8a5d-ae1ebd778bbd)


> And I don't know if Ubuntu has such a thing as a Nvidia dev package or library, but if it does, perhaps installing that too might help.

I think there are, but I'm not sure if they are necessary when building (probably not), but by default I install the Nvidia package by Ubuntu.

"QPixmap::scaled: Pixmap is a null pixmap" I guess is just the debug log from QT, and has nothing to do with it.
